### DISCUSSION TITLE
could not create a primitive

### DISCUSSION THREAD

**Discussion by oscarramirezs** [Project-MONAI/tutorials#1049] [Q&A]
hello

thanks guys for this work,,

I am working on a segmentation task using Swin UNETR, and after the  first epoch I got the following error => RuntimeError: could not create a primitive`

`
1/1, train_loss: 0.7774
epoch 1 average loss: 0.7774
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
~\AppData\Local\Temp\ipykernel_2596\3568985229.py in <module>
     77                 #roi_size = (256, 256, 192)
     78                 sw_batch_size = 1
---> 79                 val_outputs = sliding_window_inference(
     80                     val_inputs, roi_size, sw_batch_size, model)
     81                 val_outputs = [post_pred(i) for i in decollate_batch(val_outputs)]

C:\ProgramData\Anaconda3\lib\site-packages\monai\inferers\utils.py in sliding_window_inference(inputs, roi_size, sw_batch_size, predictor, overlap, mode, sigma_scale, padding_mode, cval, sw_device, device, progress, roi_weight_map, *args, **kwargs)
    178             [convert_data_type(inputs[win_slice], torch.Tensor)[0] for win_slice in unravel_slice]
    179         ).to(sw_device)
--> 180         seg_prob_out = predictor(window_data, *args, **kwargs)  # batched patch segmentation
    181 
    182         # convert seg_prob_out to tuple seg_prob_tuple, this does not allocate new memory.

C:\ProgramData\Anaconda3\lib\site-packages\torch\nn\modules\module.py in _call_impl(self, *input, **kwargs)
   1188         if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks
   1189                 or _global_forward_hooks or _global_forward_pre_hooks):
-> 1190             return forward_call(*input, **kwargs)
   1191         # Do not call functions when jit is used
   1192         full_backward_hooks, non_full_backward_hooks = [], []

C:\ProgramData\Anaconda3\lib\site-packages\monai\networks\nets\swin_unetr.py in forward(self, x_in)
    296     def forward(self, x_in):
    297         hidden_states_out = self.swinViT(x_in, self.normalize)
--> 298         enc0 = self.encoder1(x_in)
    299         enc1 = self.encoder2(hidden_states_out[0])
    300         enc2 = self.encoder3(hidden_states_out[1])

C:\ProgramData\Anaconda3\lib\site-packages\torch\nn\modules\module.py in _call_impl(self, *input, **kwargs)
   1188         if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks
   1189                 or _global_forward_hooks or _global_forward_pre_hooks):
-> 1190             return forward_call(*input, **kwargs)
   1191         # Do not call functions when jit is used
   1192         full_backward_hooks, non_full_backward_hooks = [], []

C:\ProgramData\Anaconda3\lib\site-packages\monai\networks\blocks\unetr_block.py in forward(self, inp)
    256 
    257     def forward(self, inp):
--> 258         return self.layer(inp)

C:\ProgramData\Anaconda3\lib\site-packages\torch\nn\modules\module.py in _call_impl(self, *input, **kwargs)
   1188         if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks
   1189                 or _global_forward_hooks or _global_forward_pre_hooks):
-> 1190             return forward_call(*input, **kwargs)
   1191         # Do not call functions when jit is used
   1192         full_backward_hooks, non_full_backward_hooks = [], []

C:\ProgramData\Anaconda3\lib\site-packages\monai\networks\blocks\dynunet_block.py in forward(self, inp)
    102         out = self.norm2(out)
    103         if hasattr(self, "conv3"):
--> 104             residual = self.conv3(residual)
    105         if hasattr(self, "norm3"):
    106             residual = self.norm3(residual)

C:\ProgramData\Anaconda3\lib\site-packages\torch\nn\modules\module.py in _call_impl(self, *input, **kwargs)
   1188         if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks
   1189                 or _global_forward_hooks or _global_forward_pre_hooks):
-> 1190             return forward_call(*input, **kwargs)
   1191         # Do not call functions when jit is used
   1192         full_backward_hooks, non_full_backward_hooks = [], []

C:\ProgramData\Anaconda3\lib\site-packages\torch\nn\modules\container.py in forward(self, input)
    202     def forward(self, input):
    203         for module in self:
--> 204             input = module(input)
    205         return input
    206 

C:\ProgramData\Anaconda3\lib\site-packages\torch\nn\modules\module.py in _call_impl(self, *input, **kwargs)
   1188         if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks
   1189                 or _global_forward_hooks or _global_forward_pre_hooks):
-> 1190             return forward_call(*input, **kwargs)
   1191         # Do not call functions when jit is used
   1192         full_backward_hooks, non_full_backward_hooks = [], []

C:\ProgramData\Anaconda3\lib\site-packages\torch\nn\modules\conv.py in forward(self, input)
    611 
    612     def forward(self, input: Tensor) -> Tensor:
--> 613         return self._conv_forward(input, self.weight, self.bias)
    614 
    615 

C:\ProgramData\Anaconda3\lib\site-packages\torch\nn\modules\conv.py in _conv_forward(self, input, weight, bias)
    606                 self.groups,
    607             )
--> 608         return F.conv3d(
    609             input, weight, bias, self.stride, self.padding, self.dilation, self.groups
    610         )

RuntimeError: could not create a primitive`

thank you for your help 

Oscar

**Comment by tangy5** [comment]
Is this error occurs when running model on cpu device? Based on some references, it could be a limitation of specific CPU instructions, could you try confirm, if so, can you try run it on GPU? And if the problem still exists, a sample code for reproducing the error will be helpful for the team to dig into it. 

Thanks!
